models/src/test/scala/coop/rchain/models/SortedParMapSpec.scala

```diff
@@ -80,4 +83,73 @@ class SortedParMapSpec extends FlatSpec with Matchers {
     )
     assertEqual(SortedParMap(ps), SortedParMap(ps))
   }
+
+  it should "keep keys sorted" in {
+    val sortedParMapKeys = List(
+      Par(
+        connectiveUsed = true
+      ),
+      Par(
+        bundles = Seq(
+          Bundle(
+            writeFlag = true
+          )
+        )
+      ),
+      Par(
+        connectives = Seq(
+          Connective(
+            ConnBool(
+              false
+            )
+          )
+        )
+      ),
+      Par(
+        ids = Seq(
+          GPrivate()
+        )
+      ),
+      Par()
+    )
+    val sortedParMap = SortedParMap(sortedParMapKeys.zip(Seq.fill(sortedParMapKeys.size)(Par())))
+    val keys         = sortedParMap.sortedMap.keys
+    assert(keys.toList == keys.toList.sort)
+  }
+
+  it should "preserve sortedness of all elements and the whole map for all its operations" in {
+    import coop.rchain.models.testImplicits._
+
+    forAll { (pars1: Seq[Par], pars2: Seq[Par]) =>
+      val map1 = SortedParMap(pars1.zip(pars2))
+      val map2 = SortedParMap(pars2.zip(pars1))
+      checkSorted(map1)
+      checkSorted(map2)
+      Seq(map1.headOption, map2.headOption).flatten.foreach(kv => {
```

Comment:
Try to make this test find all the 'inputs have to be sorted' bugs. I think making this say
`Seq(parts1.headOption, parts2.headOption)` should suffice for the methods already called (please call the remaining ones as needed). If not, please find a different way to test that.

Reply:
Done.